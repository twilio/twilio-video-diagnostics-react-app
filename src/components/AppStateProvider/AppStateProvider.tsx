import React, { createContext, useContext, useReducer, useEffect } from 'react';
import produce from 'immer';
import { VideoInputTest } from '@twilio/rtc-diagnostics';

export enum ActivePane {
  GetStarted,
  DeviceCheck,
  DeviceError,
  CameraTest,
  Connectivity,
  Quality,
  Results,
}

interface stateType {
  activePane: ActivePane;
  videoGranted: boolean;
  audioGranted: boolean;
  deviceError: null | Error;
  videoInputTestReport: null | VideoInputTest.Report;
}

type ACTIONTYPE =
  | { type: 'set-active-pane'; newActivePane: ActivePane }
  | { type: 'next-pane' }
  | { type: 'previous-pane' }
  | { type: 'set-devices'; devices: MediaDeviceInfo[] }
  | { type: 'set-device-error'; error: Error }
  | { type: 'set-video-test-report'; report: VideoInputTest.Report };

type AppStateContextType = {
  state: stateType;
  dispatch: React.Dispatch<ACTIONTYPE>;
};

export const initialState = {
  activePane: ActivePane.GetStarted,
  videoGranted: false,
  audioGranted: false,
  deviceError: null,
  videoInputTestReport: null,
};

export const AppStateContext = createContext<AppStateContextType>(null!);

export function useAppStateContext() {
  const context = useContext(AppStateContext);
  if (!context) {
    throw new Error('useAppStateContext must be used within an AppStateProvider');
  }
  return context;
}

/**
 * This reducer function helps with the logic used to decide the activePane,
 * as well as which panes are to be shown to the user, and when.
 */

export const appStateReducer = produce((draft: stateType, action: ACTIONTYPE) => {
  switch (action.type) {
    case 'set-active-pane':
      draft.activePane = action.newActivePane;
      break;

    case 'next-pane':
      switch (draft.activePane) {
        case ActivePane.GetStarted:
          if (draft.audioGranted && draft.videoGranted) {
            draft.activePane = ActivePane.CameraTest;
          } else {
            draft.activePane = ActivePane.DeviceCheck;
          }
          break;
        case ActivePane.DeviceCheck:
          if (draft.deviceError) {
            draft.activePane = ActivePane.DeviceError;
          } else {
            draft.activePane = ActivePane.CameraTest;
          }
          break;
        default:
          draft.activePane++;
          break;
      }
      break;

    case 'previous-pane':
      switch (draft.activePane) {
        case ActivePane.CameraTest:
          if (draft.audioGranted && draft.videoGranted) {
            draft.activePane = ActivePane.GetStarted;
          } else {
            draft.activePane = ActivePane.DeviceCheck;
          }
          break;
        default:
          draft.activePane--;
          break;
      }
      break;

    case 'set-devices':
      draft.audioGranted = action.devices.filter((d) => d.kind === 'audioinput').every((d) => d.label);
      draft.videoGranted = action.devices.filter((d) => d.kind === 'videoinput').every((d) => d.label);
      break;

    case 'set-device-error':
      draft.deviceError = action.error;
      draft.activePane = ActivePane.DeviceError;
      break;

    case 'set-video-test-report':
      draft.videoInputTestReport = action.report;
  }
});

export const AppStateProvider: React.FC = ({ children }) => {
  const [state, dispatch] = useReducer(appStateReducer, initialState);

  useEffect(() => {
    navigator.mediaDevices.enumerateDevices().then((devices) => dispatch({ type: 'set-devices', devices }));
  }, []);

  return <AppStateContext.Provider value={{ state, dispatch }}>{children}</AppStateContext.Provider>;
};
