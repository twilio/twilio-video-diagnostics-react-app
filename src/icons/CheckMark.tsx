export function CheckMark() {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M10 18C14.4183 18 18 14.4183 18 10C18 5.58172 14.4183 2 10 2C5.58172 2 2 5.58172 2 10C2 14.4183 5.58172 18 10 18ZM9.7103 12.7047L13.7062 8.71128C13.8943 8.52328 14 8.26831 14 8.00244C14 7.73657 13.8943 7.4816 13.7062 7.29361C13.5181 7.10561 13.263 7 12.9969 7C12.7309 7 12.4758 7.10561 12.2877 7.29361L9.00103 10.5882L7.71234 9.29032C7.52423 9.10232 7.2691 8.99672 7.00307 8.99672C6.73703 8.99672 6.4819 9.10232 6.2938 9.29032C6.10568 9.47831 6 9.73328 6 9.99915C6 10.265 6.10568 10.52 6.2938 10.708L8.29175 12.7047C8.38462 12.7983 8.4951 12.8725 8.61684 12.9232C8.73857 12.9739 8.86914 13 9.00103 13C9.1329 13 9.26347 12.9739 9.3852 12.9232C9.50694 12.8725 9.61742 12.7983 9.7103 12.7047Z"
        fill="#0E7C3A"
      />
    </svg>
  );
}
