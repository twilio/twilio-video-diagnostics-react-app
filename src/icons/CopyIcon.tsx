export function CopyIcon() {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M14.6812 3.59414C14.6188 2.97958 14.0998 2.5 13.4688 2.5H3.71875L3.59414 2.50629C2.97958 2.5687 2.5 3.08772 2.5 3.71875V13.4688L2.50629 13.5934C2.5687 14.2079 3.08772 14.6875 3.71875 14.6875H5.3125V16.0938C5.3125 16.8704 5.9421 17.5 6.71875 17.5H16.0938C16.8704 17.5 17.5 16.8704 17.5 16.0938V6.71875C17.5 5.9421 16.8704 5.3125 16.0938 5.3125H14.6875V3.71875L14.6812 3.59414ZM13.75 5.3125V3.71875C13.75 3.58561 13.6575 3.47408 13.5332 3.44493L13.4688 3.4375H3.71875C3.58561 3.4375 3.47408 3.53001 3.44493 3.65426L3.4375 3.71875V13.4688C3.4375 13.6019 3.53001 13.7134 3.65426 13.7426L3.71875 13.75H5.3125V6.71875C5.3125 5.9421 5.9421 5.3125 6.71875 5.3125H13.75ZM6.71875 6.25H16.0938C16.3526 6.25 16.5625 6.45987 16.5625 6.71875V16.0938C16.5625 16.3526 16.3526 16.5625 16.0938 16.5625H6.71875C6.45987 16.5625 6.25 16.3526 6.25 16.0938V6.71875C6.25 6.45987 6.45987 6.25 6.71875 6.25Z"
        fill="#606B85"
      />
    </svg>
  );
}
