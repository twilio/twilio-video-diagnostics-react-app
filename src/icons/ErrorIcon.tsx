export function ErrorIcon() {
  return (
    <div style={{ display: 'inline-block', marginRight: '10px' }}>
      <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M13.7461 1.52332L22.4767 10.2539C23.4411 11.2183 23.4411 12.7818 22.4767 13.7462L13.7461 22.4768C12.7818 23.4411 11.2182 23.4411 10.2539 22.4768L1.52327 13.7462C0.55891 12.7818 0.55891 11.2183 1.52327 10.2539L10.2539 1.52332C11.2182 0.55896 12.7818 0.55896 13.7461 1.52332ZM13.4 16.2C13.4 16.9732 12.7732 17.6 12 17.6C11.2268 17.6 10.6 16.9732 10.6 16.2C10.6 15.4269 11.2268 14.8 12 14.8C12.7732 14.8 13.4 15.4269 13.4 16.2ZM11.9856 7.80005H12.0142C12.8694 7.81042 13.5156 8.62935 13.3826 9.54159L12.7839 12.6744C12.7269 13.0891 12.3943 13.4 12.0047 13.4H11.9951C11.6055 13.4 11.2729 13.0891 11.2159 12.6744L10.6172 9.54159C10.4842 8.62935 11.1399 7.80005 11.9856 7.80005Z"
          fill="#D61F1F"
        />
      </svg>
    </div>
  );
}
