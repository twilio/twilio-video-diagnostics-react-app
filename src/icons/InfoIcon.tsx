export function InfoIcon() {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M18 10C18 5.58172 14.4183 2 10 2C5.58172 2 2 5.58172 2 10C2 14.4183 5.58172 18 10 18C14.4183 18 18 14.4183 18 10ZM3.25 10C3.25 6.27208 6.27208 3.25 10 3.25C13.7279 3.25 16.75 6.27208 16.75 10C16.75 13.7279 13.7279 16.75 10 16.75C6.27208 16.75 3.25 13.7279 3.25 10ZM9 10C9 9.44772 9.44772 9 10 9C10.5523 9 11 9.44772 11 10V13C11 13.5523 10.5523 14 10 14C9.44772 14 9 13.5523 9 13V10ZM10 8C10.5523 8 11 7.55228 11 7C11 6.44772 10.5523 6 10 6C9.44772 6 9 6.44772 9 7C9 7.55228 9.44772 8 10 8Z"
        fill="#606B85"
      />
    </svg>
  );
}
