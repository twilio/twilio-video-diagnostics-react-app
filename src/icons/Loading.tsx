export function Loading() {
  return (
    <svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M5.75469 17.1915C5.27668 17.0676 4.99236 16.5901 5.11964 16.1249C6.90138 9.61316 12.9838 5 20.0075 5C25.877 5 31.1111 8.22361 33.7061 13.1244L34.2649 7.43001L34.2946 7.27377C34.4096 6.87105 34.8017 6.58782 35.2366 6.61943C35.7259 6.65499 36.0841 7.07557 36.0368 7.5588L35.215 15.9414L35.1819 16.1099C35.0368 16.5946 34.4946 16.883 33.9987 16.699L26.0765 13.7592L25.9333 13.6909C25.5757 13.4782 25.4181 13.0341 25.5803 12.6306L25.653 12.4877C25.877 12.1303 26.3314 11.9647 26.7371 12.1153L32.2339 14.1556C29.985 9.69468 25.2869 6.74328 20.0075 6.74328C13.8006 6.74328 8.42477 10.8205 6.85066 16.5735C6.73752 16.987 6.33725 17.2516 5.91397 17.2183L5.75469 17.1915ZM34.2457 22.8085L34.0864 22.7817C33.6631 22.7484 33.2628 23.013 33.1497 23.4265C31.5756 29.1795 26.1998 33.2567 19.9929 33.2567C14.7137 33.2567 10.0158 30.3056 7.76677 25.8451L13.2633 27.885C13.6691 28.0356 14.1234 27.87 14.3475 27.5125L14.4201 27.3697C14.5823 26.9662 14.4248 26.5221 14.0672 26.3094L13.9239 26.241L6.00169 23.3013C5.50582 23.1173 4.96368 23.4057 4.81853 23.8904L4.78546 24.0589L3.96368 32.4415C3.91631 32.9247 4.27455 33.3453 4.76384 33.3809C5.19876 33.4125 5.59083 33.1292 5.70586 32.7265L5.73555 32.5703L6.29363 26.8744C8.88845 31.7759 14.1229 35 19.9929 35C27.0166 35 33.099 30.3868 34.8807 23.8751C35.008 23.4099 34.7237 22.9324 34.2457 22.8085Z"
        fill="#606B85"
      />
    </svg>
  );
}
