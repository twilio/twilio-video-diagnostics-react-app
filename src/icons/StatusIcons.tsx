export const SuccessStatus = () => (
  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <rect width="20" height="20" fill="white" fillOpacity="0.01" />
    <circle cx="10" cy="10" r="6" fill="#14B053" />
  </svg>
);

export const ErrorStatus = () => (
  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0H20V20H0V0Z" fill="white" fillOpacity="0.01" />
    <rect x="10.0713" y="3" width="10" height="10" transform="rotate(45 10.0713 3)" fill="#D61F1F" />
  </svg>
);

export const WarningStatus = () => (
  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0H20V20H0V0Z" fill="white" fillOpacity="0.01" />
    <rect x="10.0713" y="3" width="10" height="10" transform="rotate(45 10.0713 3)" fill="#F5A623" />
  </svg>
);
