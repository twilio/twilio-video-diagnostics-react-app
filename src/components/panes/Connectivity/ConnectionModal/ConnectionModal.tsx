import {
  createStyles,
  makeStyles,
  Typography,
  Dialog,
  DialogTitle,
  DialogContent,
  IconButton,
  Table,
  TableHead,
  TableBody,
  TableRow,
  TableCell,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import { ErrorStatus, SuccessStatus } from '../../../../icons/StatusIcons';

const useStyles = makeStyles((theme) =>
  createStyles({
    modal: {
      width: '100%',
    },
    content: {
      padding: '3em',
    },
    closeButton: {
      position: 'absolute',
      right: theme.spacing(1),
      top: theme.spacing(1),
      color: theme.palette.grey[500],
    },
    iconContainer: {
      display: 'flex',
      alignItems: 'center',
      '& svg': {
        marginRight: '0.3em',
      },
    },
  })
);

interface ConnectionModalProps {
  isModalOpen: boolean;
  setIsModalOpen: (isModalOpen: boolean) => void;
  serviceStatus: string;
  signalingGateway: string;
  turnServers: string;
}

export function ConnectionModal({
  isModalOpen,
  setIsModalOpen,
  serviceStatus,
  signalingGateway,
  turnServers,
}: ConnectionModalProps) {
  const classes = useStyles();

  return (
    <>
      <Dialog open={isModalOpen} onClose={() => setIsModalOpen(false)} classes={{ paper: classes.modal }}>
        <DialogTitle disableTypography>
          <Typography variant="h3">Detailed connection information</Typography>
          <IconButton aria-label="close" className={classes.closeButton} onClick={() => setIsModalOpen(false)}>
            <CloseIcon />
          </IconButton>
        </DialogTitle>
        <DialogContent dividers className={classes.content}>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>
                  <Typography variant="body1">
                    <strong>Status</strong>
                  </Typography>
                </TableCell>
                <TableCell>
                  <Typography variant="body1">
                    <strong>Connection Type</strong>
                  </Typography>
                </TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              <TableRow>
                <TableCell>
                  <div className={classes.iconContainer}>
                    {serviceStatus === 'Up' ? <SuccessStatus /> : <ErrorStatus />}
                    <Typography variant="body1">
                      <strong>{serviceStatus}</strong>
                    </Typography>
                  </div>
                </TableCell>
                <TableCell>
                  <Typography variant="body1">Twilio Services</Typography>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell>
                  <div className={classes.iconContainer}>
                    {signalingGateway === 'Reachable' ? <SuccessStatus /> : <ErrorStatus />}
                    <Typography variant="body1">
                      <strong>{signalingGateway}</strong>
                    </Typography>
                  </div>
                </TableCell>
                <TableCell>
                  <Typography variant="body1">Signaling Gateway</Typography>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell>
                  <div className={classes.iconContainer}>
                    {turnServers === 'Reachable' ? <SuccessStatus /> : <ErrorStatus />}
                    <Typography variant="body1">
                      <strong>{turnServers}</strong>
                    </Typography>
                  </div>
                </TableCell>
                <TableCell>
                  <Typography variant="body1">TURN Servers</Typography>
                </TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </DialogContent>
      </Dialog>
    </>
  );
}
