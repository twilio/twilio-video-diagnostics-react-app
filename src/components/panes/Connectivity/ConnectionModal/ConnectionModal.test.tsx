import { shallow } from 'enzyme';
import { ConnectionModal, TwilioStatusRow } from './ConnectionModal';
import { ErrorStatus, SuccessStatus, WarningStatus } from '../../../../icons/StatusIcons';
import { TwilioStatus } from '../../../AppStateProvider/AppStateProvider';

const mockServiceStatuses: TwilioStatus = {
  'Group Rooms': 'operational',
  'Go Rooms': 'operational',
  'Peer-to-Peer Rooms': 'operational',
  Recordings: 'operational',
  Compositions: 'operational',
  'Network Traversal Service': 'operational',
};

describe('the TwilioStatusRow component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<TwilioStatusRow status="operational" serviceName="Compositions" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should use the success icons when a service status is "operational"', () => {
    const wrapper = shallow(<TwilioStatusRow status="operational" serviceName="Compositions" />);
    expect(wrapper.find(SuccessStatus).exists()).toBe(true);
  });

  it('should use the error icons when a service status is major_outage', () => {
    const wrapper = shallow(<TwilioStatusRow status="major_outage" serviceName="Group Rooms" />);
    expect(wrapper.find(ErrorStatus).exists()).toBe(true);
  });

  it('should use the warning icons when a service status is partial_outage', () => {
    const wrapper = shallow(<TwilioStatusRow status="partial_outage" serviceName="Compositions" />);
    expect(wrapper.find(WarningStatus).exists()).toBe(true);
  });

  it('should use the warning icons when a service status is degraded_performance', () => {
    const wrapper = shallow(<TwilioStatusRow status="degraded_performance" serviceName="Go Rooms" />);
    expect(wrapper.find(WarningStatus).exists()).toBe(true);
  });
});

describe('the ConnectionModal component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <ConnectionModal
        isModalOpen={true}
        setIsModalOpen={jest.fn()}
        serviceStatuses={mockServiceStatuses}
        signalingGateway="Reachable"
        turnServers="Reachable"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should only use the success icons when signalingGateway and turnServers are "Reachable"', () => {
    const wrapper = shallow(
      <ConnectionModal
        isModalOpen={true}
        setIsModalOpen={jest.fn()}
        serviceStatuses={mockServiceStatuses}
        signalingGateway="Reachable"
        turnServers="Reachable"
      />
    );

    expect(wrapper.find(SuccessStatus).exists()).toBe(true);
    expect(wrapper.find(ErrorStatus).exists()).toBe(false);
  });

  it('should use the error icons when signalingGateway and turnServers are "Unreachable"', () => {
    const wrapper = shallow(
      <ConnectionModal
        isModalOpen={true}
        setIsModalOpen={jest.fn()}
        serviceStatuses={mockServiceStatuses}
        signalingGateway="Unreachable"
        turnServers="Unreachable"
      />
    );
    expect(wrapper.find(ErrorStatus).exists()).toBe(true);
  });
});
