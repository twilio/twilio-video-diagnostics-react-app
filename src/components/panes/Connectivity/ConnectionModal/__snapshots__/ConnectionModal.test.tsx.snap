// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ConnectionModal component should render correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Dialog))
    classes={
      Object {
        "paper": "makeStyles-modal-1",
      }
    }
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(DialogTitle))
      disableTypography={true}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h3"
      >
        Detailed connection information
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="close"
        className="makeStyles-closeButton-3"
        onClick={[Function]}
      >
        <Memo(ForwardRef(CloseIcon)) />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))
      className="makeStyles-content-2"
      dividers={true}
    >
      <WithStyles(ForwardRef(Table))>
        <WithStyles(ForwardRef(TableHead))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Typography))
                variant="body1"
              >
                <strong>
                  Status
                </strong>
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Typography))
                variant="body1"
              >
                <strong>
                  Type
                </strong>
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableBody))>
          <TwilioStatusRow
            serviceName="Compositions"
            status="operational"
          />
          <TwilioStatusRow
            serviceName="Go Rooms"
            status="operational"
          />
          <TwilioStatusRow
            serviceName="Group Rooms"
            status="operational"
          />
          <TwilioStatusRow
            serviceName="Network Traversal"
            status="operational"
          />
          <TwilioStatusRow
            serviceName="Peer-to-Peer Rooms"
            status="operational"
          />
          <TwilioStatusRow
            serviceName="Recordings"
            status="operational"
          />
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <SuccessStatus />
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  <strong>
                    Reachable
                  </strong>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Typography))
                variant="body1"
              >
                Signaling Gateway
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <SuccessStatus />
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  <strong>
                    Reachable
                  </strong>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Typography))
                variant="body1"
              >
                TURN Servers
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableBody))>
      </WithStyles(ForwardRef(Table))>
    </WithStyles(ForwardRef(DialogContent))>
  </WithStyles(ForwardRef(Dialog))>
</Fragment>
`;

exports[`the TwilioStatusRow component should render correctly 1`] = `
<WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableCell))>
    <div
      className="makeStyles-iconContainer-4"
    >
      <SuccessStatus />
      <WithStyles(ForwardRef(Typography))
        variant="body1"
      >
        <strong>
          Up
        </strong>
      </WithStyles(ForwardRef(Typography))>
    </div>
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      Compositions
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(TableCell))>
</WithStyles(ForwardRef(TableRow))>
`;
